core/state/statedb.go

```diff
@@ -1243,55 +1251,59 @@ func (s *StateDB) commit(deleteEmptyObjects bool) (*stateUpdate, error) {
 
 	origin := s.originalRoot
 	s.originalRoot = root
-	return newStateUpdate(origin, root, deletes, updates, nodes), nil
+
+	s.commit = newStateUpdate(origin, root, deletes, updates, nodes)
```

Comment:
Do we maybe want to make some kind of warning if `s.commit` is non-nil here, and is overwritten?